src/engraving/rw/write/twrite.cpp

```diff
@@ -434,12 +434,40 @@ void TWrite::writeSystemLocks(const Score* score, XmlWriter& xml)
     xml.endElement();
 }
 
+void TWrite::writeItemEid(const EngravingObject* item, XmlWriter& xml)
+{
+    if (!itemNeedsWriteEid(item)) {
+        return;
+    }
+
+    EID eid = item->eid();
+    if (!eid.isValid()) {
+        eid = item->assignNewEID();
+    }
+    xml.tag("eid", eid.toStdString());
+}
+
+bool TWrite::itemNeedsWriteEid(const EngravingObject* item)
```

Comment:
why not write for all elements?

Reply:
I'd like to do some performance testing on a big file. I'll come back to you about this :)